contrib/runners/orquesta_runner/tests/unit/test_rerun.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2019 Extreme Networks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import mock
+import uuid
+
+import st2tests
+
+import st2tests.config as tests_config
+tests_config.parse_args()
+
+from st2common.bootstrap import actionsregistrar
+from st2common.bootstrap import runnersregistrar
+from st2common.constants import action as ac_const
+from st2common.models.db import liveaction as lv_db_models
+from st2common.runners import base as runners
+from st2common.services import action as ac_svc
+from st2common.transport import liveaction as lv_ac_xport
+from st2common.transport import publishers
+from st2tests.mocks import liveaction as mock_lv_ac_xport
+
+
+TEST_FIXTURES = {
+    'workflows': [
+        'rerun.yaml'
+    ],
+    'actions': [
+        'rerun.yaml'
+    ]
+}
+
+TEST_PACK = 'orquesta_tests'
+TEST_PACK_PATH = st2tests.fixturesloader.get_fixtures_packs_base_path() + '/' + TEST_PACK
+
+PACKS = [
+    TEST_PACK_PATH,
+    st2tests.fixturesloader.get_fixtures_packs_base_path() + '/core'
+]
+
+OPTIONS = {
+    'tasks': ['task1']
+}
+
+
+@mock.patch.object(
+    publishers.CUDPublisher,
+    'publish_update',
+    mock.MagicMock(return_value=None))
+@mock.patch.object(
+    lv_ac_xport.LiveActionPublisher,
+    'publish_create',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_create))
+@mock.patch.object(
+    lv_ac_xport.LiveActionPublisher,
+    'publish_state',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_state))
+class OrquestRunnerTest(st2tests.WorkflowTestCase, st2tests.ExecutionDbTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(OrquestRunnerTest, cls).setUpClass()
+
+        # Register runners.
+        runnersregistrar.register_runners()
+
+        # Register test pack(s).
+        actions_registrar = actionsregistrar.ActionsRegistrar(
+            use_pack_cache=False,
+            fail_on_failure=True
+        )
+
+        for pack in PACKS:
+            actions_registrar.register_from_pack(pack)
+
+    @staticmethod
+    def get_runner_class(runner_name):
+        return runners.get_runner(runner_name).__class__
+
+    def assert_raises_with_message(self, msg, func, *args, **kwargs):
+        try:
+            self.assertRaises(func, *args, **kwargs)
+        except Exception as inst:
+            self.assertEqual(inst.message, msg)
```

Comment:
I don't believe this method is correctly implemented.  This is a generic function, please move this into the `BaseTestCase` class and add unit tests to show that this will work for catching and NOT catching the error message and error type.